src/player.cpp

```diff
@@ -5063,61 +5065,266 @@ void player::suffer()
                 }
             }
         }
-        if ( ( ( has_trait( trait_SCHIZOPHRENIC ) || has_artifact_with( AEP_SCHIZO ) ) &&
-            one_in( 2400 ) ) && is_player() ) { // Every 4 hours or so
-            monster phantasm;
-            int i;
-            switch(rng(0, 11)) {
-                case 0:
-                    add_effect( effect_hallu, 6_hours );
-                    break;
-                case 1:
-                    add_effect( effect_visuals, rng( 15_turns, 60_turns ) );
-                    break;
-                case 2:
-                    add_msg_if_player(m_warning, _("From the south you hear glass breaking."));
-                    break;
-                case 3:
-                    add_msg_if_player(m_warning, _("YOU SHOULD QUIT THE GAME IMMEDIATELY."));
-                    add_morale(MORALE_FEELING_BAD, -50, -150);
-                    break;
-                case 4:
-                    for (i = 0; i < 10; i++) {
-                        add_msg("XXXXXXXXXXXXXXXXXXXXXXXXXXX");
+        if( ( has_trait( trait_SCHIZOPHRENIC ) || has_artifact_with( AEP_SCHIZO ) ) ) {
+            if( is_player() ) {
+                bool done_effect = false;
+                // Sound
+                if( one_in( to_turns<int>( time_duration::from_hours( 4_hours ) ) ) ) {
+                    sound_hallu();
+                }
+
+                // Follower turns hostile
+                if( !done_effect && one_in( to_turns<int>( time_duration::from_hours( 4_hours ) ) ) ) {
+                    std::vector<std::shared_ptr<npc>> followers = overmap_buffer.get_npcs_near_player( 12 );
+
+                    if( !followers.empty() ) {
+                        std::string i_name = followers[rng( 0, followers.size() - 1 )]->name;
```

Comment:
There is `random_entry_ref` to get a random entry from a container: ` = random_entry_ref( followers )->name;`